pkg/cluster/internal/create/actions/config/config.go

```diff
@@ -241,6 +252,31 @@ func getKubeadmConfig(cfg *config.Cluster, data kubeadm.ConfigData, node nodes.N
 	return removeMetadata(patchedConfig), nil
 }
 
+// getKubeadmPatches generates the kubeadm patch contents. It returns a map of patch file name to patch content.
+func getKubeadmPatches(data kubeadm.ConfigData) (map[string]string, error) {
+	ver, err := version.ParseGeneric(data.KubernetesVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	patches := map[string]string{}
+	// Kubernetes older than v1.25 don't support patching kubeconfig files.
+	if ver.AtLeast(version.MustParseSemantic("v1.25.0")) {
+		// controller-manager and scheduler connect to local API endpoint, which defaults to the advertise address of the
+		// API server. If the Node's IP changes (which could happen after docker restarts), the server address in KubeConfig
+		// should be updated. However, the server certificate isn't valid for the new IP. To resolve it, we update the
```

Comment:
we should probably fix *this*, perhaps in the entrypoint we can re-roll the cert if it exists?

Reply:
Thanks for the suggestion. It reminds me that since we have the CA cert and key, we could just re-roll the cert and replace the server address in kubeconfig with the new Node IP like the existing magic: https://github.com/kubernetes-sigs/kind/blob/1a6bfa7576ad06903f72fd7c2e7f5d510296bef0/images/base/files/usr/local/bin/entrypoint#L413-L414
Then everything should just work in theory. Do you think this is a better solution?

Reply:
I tried the above approach and it worked as expected, created issue https://github.com/kubernetes-sigs/kind/issues/2759 to discuss the approaches.